connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/agent/DefaultAgentRequestHandler.java

```diff
@@ -132,9 +169,19 @@ private AgentResponse handleRequest(
         toolCalls.stream().map(ToolCallProcessVariable::from).toList();
 
     // store memory reference to context
+    LOGGER.debug("Storing runtime memory to conversation session");
     agentContext = session.storeFromRuntimeMemory(agentContext, runtimeMemory);
 
     return responseHandler.createResponse(
         executionContext, agentContext, assistantMessage, processVariableToolCalls);
   }
+
+  protected abstract boolean modelCallPrerequisitesFulfilled(
+      C executionContext, AgentContext agentContext, List<Message> addedUserMessages);
+
+  /** Handles job completion if needed. Agent response and conversation store may be null. */
```

Comment:
I wanted to suggest adding explicit nullability annotations but which ones should/do we use?

Reply:
Good point. I now added the Spring ones as we already use them in the records. For the future we could consider JSpecify as Spring will also start using them with the [next major version](https://docs.spring.io/spring-framework/reference/core/null-safety.html).